tests/api/auth/test_basic.py

```diff
@@ -0,0 +1,162 @@
+"""Tests for basic authentication for the Ralph API."""
+
+import base64
+import json
+
+import bcrypt
+import pytest
+
+from ralph.api.auth.basic import ServerUsersCredentials, UserCredentials
+from ralph.conf import settings
+
+STORED_CREDENTIALS = json.dumps(
+    [
+        {
+            "username": "ralph",
+            "hash": bcrypt.hashpw(b"admin", bcrypt.gensalt()).decode("UTF-8"),
+            "scopes": ["ralph_test_scope"],
+        }
+    ]
+)
+
+
+def test_api_auth_basic_model_serveruserscredentials():
+    """Test api.auth ServerUsersCredentials model."""
+
+    users = ServerUsersCredentials(
+        __root__=[
+            UserCredentials(
+                username="johndoe", hash="notrealhash", scopes=["johndoe_scope"]
+            ),
+            UserCredentials(username="foo", hash="notsorealhash", scopes=["foo_scope"]),
+        ]
+    )
+    other_users = ServerUsersCredentials.parse_obj(
+        [
+            UserCredentials(
+                username="janedoe", hash="notreallyrealhash", scopes=["janedoe_scope"]
+            ),
+        ]
+    )
+
+    # Test addition operator
+    users += other_users
+
+    # Test len
+    assert len(users) == 3
+
+    # Test getitem
```

Comment:
`Test getitem` is actually tested in `Test iterator`. Is it necessary to have both ?

Reply:
Behind the scene, it does tests different methods (`__getitem__`, `__iter__`), isn't it?